core/lib/instrument.c

```diff
@@ -2790,6 +2790,20 @@ dr_get_process_id(void)
     return (process_id_t)get_process_id();
 }
 
+DR_API process_id_t
+dr_get_process_id_ext(void *drcontext)
```

Comment:
Not sure about the name, I used it only because I've seen other functions instrument_api using the `foo_ext` naming when they take extra arguments compared to `foo`. Other suggestions are welcome.

Reply:
Actually they do not use `foo_ext`, they use `foo_ex`.  But this seems different. `_ex` is generally adding extra functionality or optional extra behavior on top of the same base behavior.  This is changing the base behavior.  I would suggest `dr_get_process_id_from_drcontext`.  It's longer but descriptive and clear IMHO.

Reply:
Done.